### ISSUE TITLE
How can we achieve perfect detection of large targets? How can the frame be set up correctly?

### ISSUE THREAD

**Issue by God-song** [ultralytics/ultralytics#21150]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

![Image](https://github.com/user-attachments/assets/92e9bb52-a3d6-4bd4-90a7-d79d015f8147)
Hello, recently I have been working on table positioning. The table occupies a large proportion in the entire image and is a major target. The size of the image is set at 1280*1280. The model selected is yolov8-x. However, during training, there is always a situation where the predicted boxes are one pixel smaller than the labels, and there are quite a few columns or rows that are missed. As shown in the figure, the red part is the label and the smaller part is the predicted box. How can I solve this problem?

### Additional

![Image](https://github.com/user-attachments/assets/777a3e13-8093-44b5-abf6-3ba28d15e503)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @God-song, thank you for reaching out and sharing your question with the Ultralytics team 🚀! This is an automated response to help guide you until an Ultralytics engineer can provide further assistance.

We recommend checking the [Docs](https://docs.ultralytics.com/) for detailed guidance, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions are answered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) if you haven't already. This helps us investigate and resolve the issue more efficiently.

If this is a custom training ❓ Question (such as improving detection of large objects), please include as much detail as possible—dataset examples, training logs, and confirm you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best:
- Real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions: [Discourse](https://community.ultralytics.com/)
- Share knowledge: [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

Make sure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

An Ultralytics engineer will review your question and provide more targeted guidance soon. Thank you for your patience and for contributing to improving Ultralytics!

**Comment by Y-T-G** [comment]
It's because your objects are large. For large objects, you should use smaller `imgsz`.

Or you need to increase `reg_max` by editing code:

https://github.com/ultralytics/ultralytics/blob/c132c6e47650ad1810612a6f078454d48aa18e99/ultralytics/nn/modules/head.py#L89

**Comment by God-song** [comment]
The training I'm using is 1280*1280. Do I need to adjust the training data to 960*960, or should I use the second method where only this one line needs to be modified?，self.reg_max = 16  # DFL channels (ch[0] // 16 to scale 4/8/12/16/20 for n/s/m/l/x)Change to 16？
&nbsp;


I&#39;am&nbsp;a&nbsp;pit
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Xiang ***@***.***&gt;; 
发送时间: 2025年6月23日(星期一) 下午5:53
收件人: ***@***.***&gt;; 
抄送: "I&apos;am a ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [ultralytics/ultralytics] How can we achieve perfect detection of large targets? How can the frame be set up correctly? (Issue #21150)












Y-T-G left a comment (ultralytics/ultralytics#21150)
 
It's because your objects are large. For large objects, you should use smaller imgsz.
 
Or you need to increase reg_max by editing code:
 
https://github.com/ultralytics/ultralytics/blob/c132c6e47650ad1810612a6f078454d48aa18e99/ultralytics/nn/modules/head.py#L89

**Comment by God-song** [comment]
@Y-T-G The training I'm using is 1280*1280. Do I need to adjust the training data to 960*960, or should I use the second method where only this one line needs to be modified?，self.reg_max = 16  # DFL channels (ch[0] // 16 to scale 4/8/12/16/20 for n/s/m/l/x)Change to 16？

**Comment by Y-T-G** [comment]
You change reg_max to 32 if you want to use 1280x1280

**Comment by God-song** [comment]
@Y-T-G I'm loading the yolov8-x model. I chose my own dataset. Why did I encounter this error: "Freezing layer 'model.22.dfl.conv.weight'"?

![Image](https://github.com/user-attachments/assets/126adb2a-62c3-40dd-921a-af548b61645d)

**Comment by God-song** [comment]
@Y-T-G When the number is changed from 16 to 32, with the same training data, training methods and model, why is there no change at all?

![Image](https://github.com/user-attachments/assets/45dc3e97-a194-467c-880f-02dae52aa15c)

![Image](https://github.com/user-attachments/assets/c63ed636-200b-43a9-b1c8-28eddf738205)

**Comment by Y-T-G** [comment]
You probably didn't change correctly
